### ISSUE TITLE
Update mio dependency

### ISSUE THREAD

**Issue by asafigan** [hannobraun/inotify#145]
mio version 0.6 depends on deprecated crate. Upgrade to version 0.7.

**Comment by hannobraun** [comment]
Thanks for the the note!

**Comment by ids1024** [comment]
And likewise, tokio 0.3, which has recently been released.

Updating both seems mostly easy, except for the fact Tokio no longer includes `PollEvented` in its public API (it no longer exposes it's dependency on mio in the public API in general).

I suppose the solution would be to manually implement `AsyncRead` and `AsyncWrite` similarly to how `PollEvented` does it?

**Comment by hannobraun** [comment]
Thanks, @ids1024. Any pull requests doing such upgrades are definitely welcome. Not sure what the best way to handle `PollEvented` would be. I still don't have significant experience with the whole async ecosystem (that work has always been driven by contributors).

**Comment by ids1024** [comment]
I *think* this could use [`AsyncFd`](https://github.com/tokio-rs/tokio/pull/2903) which was just merged into tokio. So I guess this is waiting on the next minor Tokio release.
